homeassistant/components/climate/generic_thermostat.py

```diff
@@ -287,23 +367,27 @@ def _async_sensor_changed(self, entity_id, old_state, new_state):
             return
 
         self._async_update_temp(new_state)
-        self._async_control_heating()
+        self._async_control_thermostat()
         yield from self.async_update_ha_state()
 
     @callback
     def _async_switch_changed(self, entity_id, old_state, new_state):
-        """Handle heater switch state changes."""
+        """Handle switch state changes."""
         if new_state is None:
             return
         self.async_schedule_update_ha_state()
 
     @callback
     def _async_keep_alive(self, time):
         """Call at constant intervals for keep-alive purposes."""
-        if self._is_device_active:
+        if self._is_heating:
```

Comment:
We should also add this here right?

```python
if self.current_operation == STATE_OFF:
    return
```

Reply:
Or even better, unsubscribing the keep alive when we're turned off.

Reply:
I'll have a look on how to achieve that